### PR TITLE
[Model and README Enhancement] Mushroom Classification using Deep Learning

### PR DESCRIPTION
**Pull request by Arihant-Bhandari**
# Pull Request for DL-Simplified 💡

## Issue Title : [Model and README Enhancement] Mushroom Classification using Deep Learning

- **Info about the related issue (Aim of the project)** : added CNN and CNN-Attention model based on Keras as well as pre-trained models
- **Name:** Arihant Bhandari
- **GitHub ID:** https://github.com/Arihant-Bhandari
- **Email ID:** arihant.bhandari.2004@gmail.com
- **Idenitfy yourself:** GSSoC 2024 Contributor

Closes: #651 [Model and README Enhancement] Mushroom Classification using Deep Learning

### Describe the add-ons or changes you've made 📃

Added 5 pre-trained models : VGG16, RESNET50, DenseNet, Xception, Inception-resnet-V2 as well as custom CNN and CNN Attention model based on Keras.

## Type of change ☑️

What sort of change have you made:

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested? ⚙️

Validation Accuracy:

VGG16: 0.2222
RESNET50: 0.0963
DenseNet: 0.1654
Xception: 0.1654
Inception-Resnet-V2: 0.1383

CNN: 0.3333 (testing accuracy: 0.3407)
CNN-Attention: 0.2444 (testing accuracy: 0.242)

## Checklist: ☑️

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by Arihant-Bhandari** [comment]
hi @abhisheks008 this is the pretrained models' work, the original author worked on similar principles hence i have followed this.

**Comment by Arihant-Bhandari** [comment]
this is the CNN and Attention mechanism work, the CNN is working better than previous implementation.

all of these models i have done are based on multi-class classification.

CNN attention performs worser than baseline CNN due to how probabilities work in favour of training, in this implementation probabilities of correct prediction are on lower end hence worse results.

**Comment by abhisheks008** [comment]
> this is the CNN and Attention mechanism work, the CNN is working better than previous implementation.
> 
> all of these models i have done are based on multi-class classification.
> 
> CNN attention performs worser than baseline CNN due to how probabilities work in favour of training, in this implementation probabilities of correct prediction are on lower end hence worse results.

Update the README file following the template, https://github.com/abhisheks008/DL-Simplified/blob/main/.github/readme_template.md

**Comment by Arihant-Bhandari** [comment]
hi @abhisheks008 pls review this pr, i have completed work currently, if there are chnages to be made i will be glad to do them; thank you for the oppurtunity and time.

**Comment by abhisheks008** [comment]
It is `README.md` not `Readme.md`. Please update it.

**Comment by Arihant-Bhandari** [comment]
> It is `README.md` not `Readme.md`. Please update it.

Part of the original author's work , will update 👍

**Comment by Arihant-Bhandari** [comment]
renamed as required @abhisheks008

**Review by abhisheks008 — APPROVED** [review]
Approved @Arihant-Bhandari
